tikv/raftstore/worker.go

```diff
@@ -244,11 +245,30 @@ func (r *splitCheckRunner) run(t task) {
 	}
 }
 
+func exceedEndKey(current, endKey []byte) bool {
+	return bytes.Compare(current, endKey) >= 0
+}
+
 /// scanSplitKeys gets the split keys by scanning the range.
 func (r *splitCheckRunner) scanSplitKeys(spCheckerHost *splitCheckerHost, region *metapb.Region,
 	startKey []byte, endKey []byte) ([][]byte, error) {
-	/// Todo, currently it is a place holder
-	return nil, nil
+	txn := r.engine.NewTransaction(false)
+	reader := dbreader.NewDBReader(startKey, endKey, txn, 0)
+	ite := reader.GetIter()
+	defer reader.Close()
+	for ite.Seek(startKey); ite.Valid(); ite.Next() {
+		item := ite.Item()
+		key := item.KeyCopy(nil)
```

Comment:
copy every key during scanSplitKeys is very expensive.
Only a few keys needed to copy.
We can avoid copy here and make sure the key is copied in split checkers

Reply:
ok, thanks.